### PR TITLE
Integrate distance threshold with FSM

### LINKED ISSUE #45: [Pod] Integrate distance threshold with FSM
Following from #11, the pod needs to stop running after exceeding the completion distance threshold.
- Add a constant for the pod's completion distance in feet
- Update the `running_periodic` transition to measure the wheel encoder position
  - If the distance travelled exceeds the threshold, transition to the `STOPPED` state

Blocked until we have the `WheelEncoder` abstraction.

### PR DESCRIPTION
**Pull request by ryescholin**
Stop the pod when reaching the completion distance threshold. Uses placeholder distance conversion and threshold values for now. Resolves #45.

### CODE REVIEW

File: pod-operation/src/state_machine.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 use std::time::Duration;
 
+use crate::components::wheel_encoder::WheelEncoder;
```
Comment (taesungh) [inline pod-operation/src/state_machine.rs:3]: Chore: could we move this import to the section after the third-party libraries (where the commented signal light import is)

File: pod-operation/src/state_machine.rs [hunk 2]
```diff
@@ -9,7 +10,8 @@ use tracing::info;
 
 // use crate::components::signal_light::SignalLight;
 
-const TICK_INTERVAL: Duration = Duration::from_millis(500);
+const TICK_INTERVAL: Duration = Duration::from_millis(10);
+const ENCODER_LIMIT: f32 = 37.0;
```
Comment (taesungh) [inline pod-operation/src/state_machine.rs:14]: Suggestion: the term "limit" isn't quite accurate as the encoder itself is not limited by such a value. Consider renaming to something like `COMPLETION_DISTANCE` or `STOP_THRESHOLD`.
Chore: for clarity, could we include an inline comment on the units of this value (i.e. feet, meters, etc.).

File: pod-operation/src/components/wheel_encoder.rs [hunk 3]
```diff
@@ -38,7 +38,7 @@ impl WheelEncoder {
 			self.counter += 1.0;
 
 			let current_time = Instant::now();
-			let distance = (self.counter * 5.0) / 1000.0;
+			let distance = (self.counter / 18.0) * 3.28084;
```
Comment (taesungh) [inline pod-operation/src/components/wheel_encoder.rs:41]: Question: could you elaborate where these magic values come from? The encoder we're using has 16 pulses per cycle (not 18), and the wheel diameter we used last year was 3 inches.

**Review by taesungh — COMMENTED** [review]
Nice start, some minor suggestions and a question about the conversion factor.

File: pod-operation/src/components/wheel_encoder.rs (hunk 3 above)
Comment (ryescholin) [reply pod-operation/src/components/wheel_encoder.rs:41]: testing values not accurate, will change now thx

**Review by taesungh — APPROVED** [review]
Works for now, will revisit the actual counter-to-distance conversion as part of #58. Please **squash** during merge.

**Review by vrushang1234 — APPROVED** [review]
Looks good! Thanks for working on it

**Pull request merged**